integration/common.go

```diff
@@ -17,38 +17,51 @@ import (
 	"github.com/stretchr/testify/suite"
 )
 
-type fileUploadSuite struct {
+// FileUploadSuite tests the file upload functionalities, using a provided configuration
+type FileUploadSuite struct {
 	suite.Suite
 	util.SuiteInitializer
 
-	thingURL   string
-	featureURL string
-	uploadCfg  uploadTestConfig
+	ThingURL   string
+	FeatureURL string
+
+	uploadCfg uploadTestConfig
+	provider  storageProvider
 }
 
 type httpFileUploadSuite struct {
-	fileUploadSuite
+	FileUploadSuite
 }
 
 type azureFileUploadSuite struct {
-	fileUploadSuite
+	FileUploadSuite
 }
 
 type awsFileUploadSuite struct {
-	fileUploadSuite
+	FileUploadSuite
 }
 
+// uploadTestConfig contains the test configuration data, such as upload directory and http server URL
 type uploadTestConfig struct {
 	UploadDir  string `env:"FUT_UPLOAD_DIR"`
 	HTTPServer string `env:"FUT_HTTP_SERVER"`
 }
 
-type upload interface {
+// storageProvider contains file upload, download and remove logic for different storage providers(i.e Azure, AWS, generic)
+type storageProvider interface {
 	requestUpload(correlationID string, filePath string) map[string]interface{}
+	downloadURL(correlationID string) (string, error)
 	download(correlationID string) ([]byte, error)
 	removeUploads()
 }
 
+// Constants for different storage providers
+const (
+	AzureStorageProvider   = "Azure"
```

Comment:
`iota` should be used instead of these strings. That's the Go way for enumerations.